### PR TITLE
[tutorial] embedded spa pattern

### PR DESCRIPTION
**Pull request by sarah11918**
#### Description (required)

New tutorial page that walks through embedding a SPA (the [Remix Run basic React Router Example](https://github.com/remix-run/react-router/tree/dev/examples)) into an empty Astro project.

Note:

- open for edits!
- TBA if this fits in the "Tutorial" category:
	- PRO: This will be part of the "v4 story" so putting it here gives it more prominence and an appropriate "onboarding" spot if we think we'll be snagging those looking for a CRA replacement
	- PRO: It *is* in tutorial format, including dictating/providing the starting code and even one multiple-choice quiz
	- PRO: It *is* at the same beginner-friendly level as the other tutorial content. It is super-easy to follow and requires very little background knowledge

	- CON??: Is this really such early-on knowledge that it belongs even before Basics?? The tutorial assists those who aren't even ready for basics yet. This is arguably way shorter, and even way easier than the Build a Blog tutorial... but only if you stick to the carefully-chosen code provided. In that sense, it does fit here. BUT, if you try to add any old CRA into Astro... you'll probably have to do some troubleshooting etc. This is my only reservation with placement, but it's not a strong one. Ultimately, even the build a blog tutorial is meant for you to get a feel what it's like, and prepare you for reading the docs. I do feel this does the same.

### CODE REVIEW

File: src/content/docs/en/tutorials/embedded-spa.mdx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+		<title>Astro</title>
+	</head>
+	<body>
+		<h1>Astro</h1>
+		<Main client:only="react"/> 
+	</body>
+</html>
+```
+
+Now, when you navigate to the root of your Astro project in the browser, you will see a fully-functional embedded SPA using React Router to control its own routing.
```
Comment (sarah11918) [inline src/content/docs/en/tutorials/embedded-spa.mdx:240]:
```suggestion
Now, when you navigate to the root of your Astro project in the browser, you will see a fully-functional embedded SPA using React Router to control its own routing.

## Next Steps

Note that more complex Apps may require additional troubleshooting. Read more about [using UI framework components](/en/core-concepts/framework-components/) in Astro for more advanced patterns and usage.
```

File: src/content/docs/en/tutorials/embedded-spa.mdx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+And as a **multi-page application**, Astro can also generate additional pages and API routes in Astro for more traditional page content at other routes... or even more embedded SPAs! This powerful embedded SPA pattern allows you to use one Astro project to house:
+
+- An app alongside your blog, or portfolio.
+- An eCommerce shop for a collection of apps, each with their own documentation and working demo.
+- Any combination of traditional web site/app pages and interactive framework components or full-fledged running apps!
+
+Over time, you may find that instead of simply embedding, you want to migrate a SPA [(e.g. Create-React-App)](/en/guides/migrate-to-astro/from-create-react-app/) by rewriting some or all of it as an Astro app. The embedded SPA pattern allows you to update incrementally, first getting your app running in your Astro project, and then refactoring to leverage the benefits of working in Astro.
+
+<Box icon="question-mark">
+### Test your knowledge
```
Comment (sarah11918) [inline src/content/docs/en/tutorials/embedded-spa.mdx:42]:
```suggestion
<h3>Test your knowledge</h3>
```

File: src/content/docs/en/tutorials/embedded-spa.mdx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+Astro can render an **embedded single-page application (SPA)** from any UI framework as an [Astro island](/en/concepts/islands/). This allows a JavaScript app to run entirely at the root of an Astro project in just a few lines of code. 
+
+```astro title="src/pages/index.astro"
+---
+// Import your root App component
+import App from '../cra-project/App.jsx'
+---
+<!-- Use a client directive to load your app -->
+<App client:load />
```
Comment (florian-lefebvre) [inline src/content/docs/en/tutorials/embedded-spa.mdx:28]: Shouldn't it be this?
```suggestion
<App client:only="react" />
```
SPAs often rely on browser APIs

**Comment by sarah11918** [comment]
I am closing this PR (but not deleting this branch) because I now think we'll probably incorporate this into the CRA page directly. And if we do a separate, visible "tutorial" item, it will be focused on *building* a tiny app to embed, then embedding it alongside. I think the focus shift is now towards putting the "bring your own" advice in the migration guides.

**Pull request closed without merging**